src/main/java/com/synopsys/integration/detect/lifecycle/run/step/RlScanStepRunner.java

```diff
@@ -140,7 +143,21 @@ private boolean isBlackDuckVersionValid() {
     }
     
     private boolean isReversingLabsEligible() {
-        return operationRunner.getRlScanFilePath().isPresent();
+        if (!operationRunner.getRlScanFilePath().isPresent()) {
+            logger.info("No detect.rl.scan.file.path property was provided. Skipping ReversingLabs scan.");
+            return false;
+        }
+        
+        String scanFilePath = operationRunner.getRlScanFilePath().get();
+        
+        File scanFile = new File(scanFilePath);
+        
+        if (!Files.isReadable(scanFile.toPath())) {
+            logger.info("Unable to access file: " + scanFilePath  + ". Please ensure the file exists and is readable by Detect.");
+            return false;
+        }
```

Comment:
I'm wondering if this needs to throw an exception and log an error? If it doesn't error out, the final output would show the status as `RL Scan: SUCCESS` at the end of Detect's run. Else, it will show the status as FAILURE because `publishRlFailure` in the catch block can catch it. Just bringing this up in case it is desired - but either way looks good to approve.

Reply:
Thanks, this is better than what I have now. I basically just added the exception bit as publishRlFailure catches it and does the logging already.